src/utils/logging.js

```diff
@@ -21,6 +23,42 @@ function withScope<R>(callback: Scope => R): R {
   return ((ret: $FlowFixMe): R);
 }
 
+/**
+ * Get server-version tags at various levels of granularity.
+ */
+const getServerVersionTags = () => {
+  const zulipVersion = tryGetActiveAccount(store.getState())?.zulipVersion;
+
+  // If there's no active account, we don't know the server version.
+  if (zulipVersion) {
+    const raw = zulipVersion.raw();
+
+    const OMITTED = 'x';
+    const UNKNOWN = '?';
+
+    const elements = zulipVersion.elements();
+    const major = elements.major ?? UNKNOWN;
+    const minor = elements.minor ?? UNKNOWN;
+    const patch = elements.patch ?? UNKNOWN;
+
+    let coarseServerVersion;
+    let fineServerVersion;
+    // Effective with 3.0, we changed our numbering conventions; 3.x and
+    // 4.x are each the same level of granularity as 2.1.x or 2.0.x.
+    if (zulipVersion.isAtLeast('3.0')) {
+      coarseServerVersion = [major, OMITTED].join('.');
+      fineServerVersion = [major, minor].join('.');
+    } else {
+      coarseServerVersion = [major, minor, OMITTED].join('.');
+      fineServerVersion = [major, minor, patch].join('.');
+    }
+
+    return { rawServerVersion: raw, coarseServerVersion, fineServerVersion };
+  } else {
+    return {};
+  }
```

Comment:
Let's make this an early return, to keep the main/happy/interesting path on a straight line. So like:
```js
  if (!zulipVersion) {
    return {};
  }

  const raw = zulipVersion.raw();
  // …
```